### PR TITLE
[14.0][ADD] base_report_to_label_printer: Add default label printer

### PR DESCRIPTION
**Pull request by ne1800**
We have the case in the company that we have to print specific reports to a label printer. This should be done via the user printer. At the user it should be possible to set a label printer and a normal printer. At the report itself it should be defined if it is a label report.

This extension could perhaps help others.

### CODE REVIEW

File: base_report_to_printer/models/res_users.py [hunk 1]
```diff
@@ -23,6 +23,9 @@ def _user_available_action_types(self):
     printing_printer_id = fields.Many2one(
         comodel_name="printing.printer", string="Default Printer"
     )
+    default_label_printer_id = fields.Many2one(
```
Comment (jbaudoux) [inline base_report_to_printer/models/res_users.py]: That new field should also be added in `_register_hook` below.
We must allow the user to set a value in his preferences

File: base_report_to_printer/models/ir_actions_report.py [hunk 2]
```diff
@@ -59,9 +60,13 @@ def print_action_for_report_name(self, report_name):
     def _get_user_default_print_behaviour(self):
         printer_obj = self.env["printing.printer"]
         user = self.env.user
+        if self.label:
```
Comment (jbaudoux) [inline base_report_to_printer/models/ir_actions_report.py]: move this in a function `_get_user_default_printer`

**Review by jbaudoux — CHANGES_REQUESTED** [review]
It is better to add this in a new module `base_report_to_label_printer`
Just implement `_get_user_default_printer` in this module and extend it in the new module.

Can you also open a PR on https://github.com/OCA/wms/tree/14.0/shopfloor_workstation_label_printer
to depend on the new module and drop from `https://github.com/OCA/wms/blob/14.0/shopfloor_workstation_label_printer/models/res_users.py` what you moved here

**Comment by jbaudoux** [comment]
cc @simahawk @sebalix

**Review by jbaudoux — APPROVED** [review]
LGTM

Can you squash the commits?

File: base_report_to_printer/models/ir_actions_report.py [hunk 3]
```diff
@@ -56,12 +56,16 @@ def print_action_for_report_name(self, report_name):
         }
         return serializable_result
 
+    def _get_user_default_printer(self, user):
+        return user.printing_printer_id
```
Comment (sebalix) [inline base_report_to_printer/models/ir_actions_report.py:60]:
```suggestion
        self.ensure_one()
        return user.printing_printer_id
```
As we will check `self.label` in the override.

**Review by sebalix — APPROVED** [review]
One small remark, LG

**Comment by jbaudoux** [comment]
cc @lmignon

**Review by TDu — APPROVED** [review]

**Comment by simahawk** [comment]
@ne1800 can you rebase pls?

File: base_report_to_printer/models/ir_actions_report.py (hunk 3 above)
Comment (simahawk) [reply base_report_to_printer/models/ir_actions_report.py:60]: @ne1800 ping

**Comment by mt-software-de** [comment]
> @ne1800 can you rebase pls?

@ne1800 is not available right now. 
I added his PR here https://github.com/jbaudoux/report-print-send/pull/2  maybe we can merge then both
@jbaudoux fyi

**Comment by jbaudoux** [comment]
Included in https://github.com/OCA/report-print-send/pull/310
This PR can be closed

**Comment by sebalix** [comment]
@mt-software-de @jbaudoux we should not mix these two PRs to separate concerns, I'm using this one in one project but not the other.
If @ne1800 is not available I can re-open this one elsewhere so it can be merged ASAP.

**Comment by jbaudoux** [comment]
> @mt-software-de @jbaudoux we should not mix these two PRs to separate concerns, I'm using this one in one project but not the other. If @ne1800 is not available I can re-open this one elsewhere so it can be merged ASAP.

Let's approve and merge the other PR, then you can use the module you want without relying on open PR
I need to include this PR in the other one for the glue module

**Comment by simahawk** [comment]
> > @mt-software-de @jbaudoux we should not mix these two PRs to separate concerns, I'm using this one in one project but not the other. If @ne1800 is not available I can re-open this one elsewhere so it can be merged ASAP.
> 
> Let's approve and merge the other PR, then you can use the module you want without relying on open PR I need to include this PR in the other one for the glue module

@jbaudoux hmm sounds like the contrary.... we must merge the base PR 1st, which is this one.
We can take this over and finish it.

**Comment by simahawk** [comment]
there you go https://github.com/OCA/report-print-send/pull/313

**Pull request closed without merging**